### PR TITLE
Admins can upload a book cover

### PR DESCRIPTION
**Pull request by stephaniewillhide**
#8

### CODE REVIEW

**Comment by stephaniewillhide** [comment]
@jordanmaguire I've added a bit of simple styling onto the drag and drop like so.

Screen Shot 2019-08-20 at 9 16 07 PM

**Comment by jordanmaguire** [comment]
@stephaniewillhide Could you merge #11 into master and then merge master into this branch?

That way the diff won't include the half-done search work.

**Comment by stephaniewillhide** [comment]
Indeed! Master has been merged.

File: app/views/books/_book.html.erb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <tr>
   <td class="px-3"><%= book.name %></td>
   <td class="px-3"><%= book.isbn %></td>
+  <td><img class="m-2 img-thumbnail" width="150" src="<%= url_for(book.cover) rescue nil %>" /></td>
```
Comment (jordanmaguire) [inline app/views/books/_book.html.erb]: I have a few recommendations here.

1. No need to attempt to show the image if we know there is none:

```erb
  <td>
    <% if book.cover.attached? %>
      <img class="m-2 img-thumbnail" width="150" src="<%= url_for(book.cover) %>" />
    <% end %>
  </td>
```

2. Use Rails view helpers where possible, to allow Rails to handle some of the lower level details. In this case we have `image_tag` at our disposal:

```erb
  <td>
    <% if book.cover.attached? %>
      <%= image_tag(book.cover, class: "m-2 img-thumbnail", width: "150") %>
    <% end %>
  </td>
```

Notice that we no longer need to generate a url for the image, we can simply pass `book.cover` into the `image_tag`.

File: app/views/books/_book.html.erb [hunk 2]
```diff
@@ -1,6 +1,13 @@
 <tr>
   <td class="px-3"><%= book.name %></td>
   <td class="px-3"><%= book.isbn %></td>
+  <td>
+    <% if book.cover.attached? %>
+      <%= image_tag(book.cover, class: "m-2 img-thumbnail", width: "150") %>
+    <% else %>
+      <%= image_tag("sample-book-cover.jpg", class: "m-2 img-thumbnail", width: "150") %>
```
Comment (jordanmaguire) [inline app/views/books/_book.html.erb]: Good call 👍 

If you want to reduce the duplication of the `image_tag` and its arguments, you could do the following:

```ruby
<% book_cover = book.cover.attached? ? book.cover : "sample-book-cover.jpg" %>
<%= image_tag(book_cover, class: "m-2 img-thumbnail", width: "150") %>
```

**Pull request merged**